src/util.c

```diff
@@ -1969,7 +1969,7 @@ nats_ReadFile(natsBuffer **buffer, int initBufSize, const char *fn)
 
     if (s == NATS_OK)
     {
-        natsBuf_AppendByte(buf, '\0');
+        s = natsBuf_AppendByte(buf, '\0');
```

Comment:
Thanks for the catch. However, if we do get the status, then we need to return buffer only if the last append is OK, otherwise, we need to destroy the buffer. So we could replace with (replacing the above `if (s == NATS_OK)`):
```
IFOK(s, natsBuf_AppendByte(buf, '\0'));

if (s == NATS_OK)
{
    *buffer = buf;
}
else if (buf != NULL)
{
...
```